run_project_tests.py

```diff
@@ -1247,11 +1247,16 @@ def _run_tests(all_tests: T.List[T.Tuple[str, T.List[TestDef], bool]],
                 testname += f' ({t.name})'
             should_fail = ''
             suite_args = []
+            fail_file = t.path / 'FAIL'
             if name.startswith('failing'):
                 should_fail = name.split('failing-')[1]
-            if name.startswith('warning'):
+            elif name.startswith('warning'):
                 suite_args = ['--fatal-meson-warnings']
                 should_fail = name.split('warning-')[1]
+            elif fail_file.exists():
+                should_fail = fail_file.read_text().strip()
+                if should_fail not in ('meson', 'build', 'test'):
+                    raise RuntimeError(f'"{fail_file}" must contain `meson`, `build`, or `test`')
```

Comment:
We already have a test description in the form of `test.json`, and that should be extended instead of adding a new file with a custom format.